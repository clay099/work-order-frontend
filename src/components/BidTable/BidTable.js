import React from "react";
import { useSelector } from "react-redux";
import TableHeadings from "../TableHeadings/TableHeadings";
import BidTableContents from "../BidTableContents/BidTableContents";
import { Table, Typography, Paper, TableContainer } from "@material-ui/core/";
import useStyles from "./styles";

/** BidTable Component
 * @param  {int} {projectId}
 *
 * Gets te Redux state for bids on the project.
 *
 * If there is no bids for this project returns null - no table is rendered
 *
 * If there is one or more bids displays a table which contains all the current bids on the project and the ability for the user to select a bid
 */
const BidTable = ({ projectId }) => {
	const classes = useStyles();
	const { bids } = useSelector((st) => ({
		bids: st.bids[projectId],
	}));

	let headingList = ["Tradesmen First Name", "Tradesmen Last Name", "Bid"];
	return (
		<>
			{bids ? (
				<div>
					<TableContainer component={Paper} className={classes.table}>
						<Typography
							component="h2"
							variant="h6"
							color="primary"
							gutterBottom
						>
							Project Bids
						</Typography>
						<Table size="small" dense table>
							<TableHeadings headings={headingList} />
							<BidTableContents
								bidData={bids}
								projectId={projectId}
							/>
						</Table>
					</TableContainer>
				</div>
			) : null}
		</>
	);
};

export default BidTable;
