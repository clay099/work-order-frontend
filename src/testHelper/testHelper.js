// testHelper.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// Import your own reducer
import reducer from "../reducers/rootReducer";

function render(
	ui,
	{
		initialState,
		store = createStore(reducer, initialState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const tokenState = { token: "token" };
const userTypeUserState = { userType: "user" };
const userTypeTradesmenState = { userType: "user" };
const idState = { id: 1234 };
const userDetailsState = { userDetails: "userDetails" };
const handleSubmitState = { handleSubmit: jest.fn() };
const handleChangeState = { handleChange: jest.fn() };
const disabledState = { disabled: false };
const disabledEmailState = { disabledEmail: false };
const buttonTextState = { buttonText: "buttonText" };
const formDataState = {
	formData: {
		id: 1,
		firstName: "firstName",
		lastName: "lastName",
		email: "email",
		password: "password",
		newPassword: "newPassword",
		confirmNewPassword: "confirmNewPassword",
	},
};

// re-export everything
export * from "@testing-library/react";
// override render method
export {
	render,
	tokenState,
	userTypeUserState,
	userTypeTradesmenState,
	idState,
	userDetailsState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
};
