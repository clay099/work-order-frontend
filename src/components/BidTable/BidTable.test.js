import React from "react";
import BidTable from "./BidTable";
import { render, bidData } from "../../testHelper/testHelper";
console.error = jest.fn();

const params = {
	projectId: 1,
};

const initialState = {
	login: { token: "token" },
	bids: {
		1: { first_name: "first_name", last_name: "last_name", bid: 500 },
	},
};

describe("BidTable Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<BidTable {...params} />, {
			initialState,
		});
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(<BidTable {...params} />, {
			initialState,
		});
		expect(asFragment).toMatchSnapshot();
	});

	test("value to be in the document", () => {
		const { getAllByText } = render(<BidTable {...params} />, {
			initialState,
		});
		const value = getAllByText("Accept Bid");
		expect(value[0]).toBeInTheDocument();
	});
});
