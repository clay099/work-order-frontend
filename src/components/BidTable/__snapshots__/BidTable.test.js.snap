// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidTable Component matches snapshot 1`] = `[Function]`;
